src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -756,13 +756,36 @@ std::string ContextAttribute::render
 *        the code paths of the rendering process
 *
 */
-std::string ContextAttribute::toJson(bool isLastElement, const std::string& renderMode, RequestType requestType)
+std::string ContextAttribute::toJson(bool isLastElement, const std::string& renderMode, NotificationFormat notifyFormat, RequestType requestType)
 {
   std::string  out;
+  bool         uniqueValues = false;
+  std::string  rMode        = renderMode;  // renderMode is 'const' and cannot be modified
+
+  //
+  // FIXME PR: To be discussed during PR-review
+  //   'notifyFormat' overrides 'renderMode' if NGSI_V2_KEYVALUES or NGSI_V2_VALUES but 'uniqueValues' must be saved
```

Comment:
Unifying sound good... However, I don't have the full context of this and not sure if this would be a good or bad idea from the point of view of complexity.

(Sorry for providing such poor feedback :( )